### ISSUE TITLE
RFC: Move quick-lint to GitLab

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#777]
The [SFConservancy](https://sfconservancy.org/) has recently sent out a memo advising developers to “[Give Up GitHub](https://sfconservancy.org/GiveUpGitHub/)” over non-free practices.

quick-lint(’s creator) — not wanting to be left behind on any trends 😎— has decided to consider moving to a Free platform.  The current contender as a replacement is GitLab.org

Pros:
- Free[^1]
- Better
- GitLab CI > GitHub Actions (objectively)

Cons:
- Community
- must migrate GitHub Actions test suit to GitLab CI (even though GitLab CI is non-debatably better)

quick-lint is a community project.  Another core feature of quick-lint is how it tries to bring in new developers to mentor them into better developers (such as through our [hiring](quick-lint-js.com/hiring) program).  While GitLab may be freer and better, the fact is most devs are on GitHub.  We shouldn’t underestimate the impact of such a migration

[^1]: Technically only the self-hosted version is free because of GitLab’s proprietary hosting

**Comment by vegerot** [comment]
Other alternatives:

- Gitea 👎: no CI.  Could use Gitea with GitLab CI

**Comment by vegerot** [comment]
We can vote by reacting to the original post

👍: GitLab
👎: GitHub
😁: Other
😕: I don’t have an opinion but I think reacting to posts is fun

**Comment by strager** [comment]
> The current contender as a replacement is [GitLab.org](https://gitlab.org/)

No. GitLab.org is proprietary in the same way that GitHub is proprietary. I know you point this out in the footnote, but it's worth emphasizing.

**Comment by vegerot** [comment]
I updated the message to remove specifically choosing GitLab.org

**Comment by vegerot** [comment]
On a related note: the community issues I pointed out are only exacerbated by using a self-hosted instance of GitLab

Many devs already have a GitLab account, and if not it's easy enough to make one (you can even sign in with GitHub in one click)

**Comment by strager** [comment]
GitHub services we currently use:

* Git storage
* Git UI
* issues
* pull requests
* CI
* analytics
* Linux CI workers
* macOS CI workers
* Windows CI workers
* Docker image hosting
* Dependabot security checks

I think self-hosted GitLab provides all of these services except the CI workers, Dependabot, and maybe Docker image hosting.

**Comment by vegerot** [comment]
Are you referring to the free or paid self-hosted GitLab?

The AF used GitLab's container registry and managed updates through [Renovate](https://docs.renovatebot.com/).

If you're self-hosting, you can use whatever workers you physically own.  How would self-hosted GitLab "provide" you with any workers?

**Comment by strager** [comment]
> Are you referring to the free or paid self-hosted GitLab?

GitLab CE.

>  How would self-hosted GitLab "provide" you with any workers?

It doesn't, and I didn't claim it would. I claimed the opposite in fact.

> If you're self-hosting, you can use whatever workers you physically own.

I know. And I can rent or buy more if needed.

**Comment by vegerot** [comment]
Gotcha.  That wasn't a rhetorical question.  I was worried I didn't understand what you meant.  But sounds like we're on the same page 👍🏿

**Comment by vegerot** [comment]
Since quick-lint is a community project, you should create a botnet of CI runners from your viewers

**Comment by strager** [comment]
quick-lint-js is not a community project.

**Comment by vegerot** [comment]
> quick-lint-js is not a community project.

What even is a community project?  I always hear “community project” on repos that encourage contributions from its users
